internal/controllers/app_controller.go

```diff
@@ -270,12 +290,245 @@ func (r *AppReconciler) reconcile(ctx context.Context, app *ketchv1.App, logger
 		}
 	}
 
+	if len(app.Spec.Deployments) > 0 {
+		// use latest deployment and watch events for each process
+		latestDeployment := app.Spec.Deployments[len(app.Spec.Deployments)-1]
+		for _, process := range latestDeployment.Processes {
+			var dep appsv1.Deployment
+			if err := r.Get(ctx, client.ObjectKey{
+				Namespace: framework.Spec.NamespaceName,
+				Name:      fmt.Sprintf("%s-%s-%d", app.GetName(), process.Name, latestDeployment.Version),
+			}, &dep); err != nil {
+				return reconcileResult{
+					status:  v1.ConditionFalse,
+					message: fmt.Sprintf("failed to get deployment: %v", err),
+				}
+			}
+			err = r.watchDeployEvents(ctx, app, framework.Spec.NamespaceName, &dep, &process, r.Recorder)
```

Comment:
Good call. I put all of the Event `watch` code in a goroutine. I was struggling a lot with an issue where the reconciler would complete before creating all events, but realized it was fixable with a `requeue` parameter rather than blocking _everything_.